sim/src/tlv.rs

```diff
@@ -267,8 +267,7 @@ impl ManifestGen for TlvGen {
             protected_tlv.push(0x69);
             protected_tlv.write_u16::<LittleEndian>(self.protect_size()).unwrap();
             for dep in &self.dependencies {
-                protected_tlv.push(TlvKinds::DEPENDENCY as u8);
-                protected_tlv.push(0);
+                protected_tlv.write_u16::<LittleEndian>(TlvKinds::DEPENDENCY as u16).unwrap();
                 protected_tlv.push(12);
                 protected_tlv.push(0);
```

Comment:
Those ordered byte pushes read much better; I would suggest you to also change the `len` pushes to use it.

Reply:
I seem to have missed all of the ones that are just hardcoded values. I did change the lengths that come from a value. I'll make a commit that changes the rest as well.